### PR TITLE
init myaccount changes - popupSlice, myaccountSlide, add pd finalization

### PR DESCRIPTION
**Pull request by iwonakulacz**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -24,7 +24,8 @@ module.exports = {
     'prettier/prettier': ['error', { singleQuote: true, endOfLine: 'auto' }],
     'react/jsx-filename-extension': ['error', { extensions: ['.js', '.jsx'] }],
     'import/no-cycle': 'off', // TODO: remove after clean localstorage
-    'react/forbid-prop-types': 'off' // TODO: remove after moving to redux
+    'react/forbid-prop-types': 'off', // TODO: remove after moving to redux
+    'react/jsx-props-no-spreading': 'off'
```
Comment (iwonakulacz) [inline .eslintrc.js:28]: are you sure, that we want to switch it off in the whole project?

File: src/components/PaymentMethod/PaymentMethod.js [hunk 2]
```diff
@@ -64,8 +64,8 @@ const PaymentMethod = ({
               title={t('Add a payment method!')}
               subtitle={t('Set up a new payment method for your account')}
               withBorder
-              onClick={() =>
-                showInnerPopup({ type: POPUP_TYPES.paymentDetails })
+              onClick={
+                () => showInnerPopup({ type: POPUP_TYPES.paymentDetails }) // TODO: cover in 3DS
```
Comment (iwonakulacz) [inline src/components/PaymentMethod/PaymentMethod.js:68]: this also should be covered :) 
It's a case when user doesn't have any payment method

File: src/components/Popup/Popup.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /* eslint-disable no-debugger */
```
Comment (iwonakulacz) [inline src/components/Popup/Popup.js:1]: can be deleted

File: src/components/Popup/Popup.js [hunk 4]
```diff
@@ -23,139 +24,114 @@ import {
...
-    });
-    this.checkAccess(consents);
-  }
-
-  renderNextStep = () => {
-    this.setState(prevState => {
-      return { step: prevState.step + 1 };
-    });
+  const checkAccess = items => {
+    const notCheckedTerm = items.find(
```
Comment (iwonakulacz) [inline src/components/Popup/Popup.js:39]: we could rename it to more meaningful, eg. `shouldBlockConfirmButton` / `requiredConsentsToAccept` or something similar

File: src/components/UpdatePaymentDetailsPopup/Steps/Error.js [hunk 5]
```diff
@@ -21,24 +37,26 @@ const Error = ({ hideInnerPopup }) => {
...
+              again using different payment method.
+            </Trans>
+          )}
         </TextStyled>
       </ContentStyled>
       <ButtonWrapperStyled removeMargin>
-        <Button theme="simple" onClickFn={() => hideInnerPopup()}>
-          {t('Back to Payment Details')}
+        <Button theme="simple" onClickFn={handleTryAgain}>
+          {t('Try again')}
```
Comment (iwonakulacz) [inline src/components/UpdatePaymentDetailsPopup/Steps/Error.js:55]: did you test the default case when this popup was presented? Eg. try to add a regular card as a payment method and provide wrong CVC. The copy will be updated to the new one and also button will be replaced but I think it's a better behavior then what we had previously

File: src/components/UpdatePaymentDetailsPopup/Steps/FinalizeAddPaymentDetails.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        dispatch(
+          updatePaymentDetailsPopup({
+            step: PAYMENT_DETAILS_STEPS.SUCCESS,
+            isLoading: false
+          })
+        );
+      })
+      .catch(() => {
+        dispatch(
+          updatePaymentDetailsPopup({ step: PAYMENT_DETAILS_STEPS.ERROR }) // TODO: customize texts
```
Comment (iwonakulacz) [inline src/components/UpdatePaymentDetailsPopup/Steps/FinalizeAddPaymentDetails.js:37]: TODO to remove :)

File: src/components/UpdatePaymentDetailsPopup/Steps/FinalizeAddPaymentDetails.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        );
+      });
+  };
+
+  useEffect(() => {
+    submitRedirectResult();
```
Comment (iwonakulacz) [inline src/components/UpdatePaymentDetailsPopup/Steps/FinalizeAddPaymentDetails.js:43]: Lets add a if: 
```
if(adyenRedirectResult && paymentMethodId) {
   submitRedirectResult();
}
```

File: src/components/UpdatePaymentDetailsPopup/Steps/Success.js [hunk 8]
```diff
@@ -23,20 +26,17 @@ const Success = ({ hideInnerPopup }) => {
         <TextStyled>{t('Your payment details have been updated')}</TextStyled>
       </ContentStyled>
       <ButtonWrapperStyled removeMargin>
-        <Button theme="simple" onClickFn={() => hideInnerPopup()}>
+        <Button
+          theme="simple"
+          onClickFn={() =>
+            dispatch(updatePaymentDetailsPopup({ isOpen: false }))
```
Comment (iwonakulacz) [inline src/components/UpdatePaymentDetailsPopup/Steps/Success.js:32]: just wondering if we should 'reset' more of the state here. What if someone immediately opens the popup once again?

File: src/components/UpdatePaymentDetailsPopup/UpdatePaymentDetailsPopup.js [hunk 9]
```diff
@@ -98,6 +100,9 @@ const UpdatePaymentDetailsPopup = ({
         }
       });
     }
+    return () => {
+      dispatch(resetPaymentDetailsPopupState());
```
Comment (iwonakulacz) [inline src/components/UpdatePaymentDetailsPopup/UpdatePaymentDetailsPopup.js:104]: regarding above comment -> probably this fixes the case when user immediately opens the popup again

File: src/containers/AddPaymentDetailsFinalizationHandler/AddPaymentDetailsFinalizationHandler.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
+import React, { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import {
+  updatePaymentDetailsPopup,
+  PAYMENT_DETAILS_STEPS
+} from 'redux/popupSlice';
+import { setActiveTab, MYACCCOUNT_TABS } from 'redux/myaccountSlice';
+
+const AddPaymentDetailsFinalizationHandler = Component => {
```
Comment (iwonakulacz) [inline src/containers/AddPaymentDetailsFinalizationHandler/AddPaymentDetailsFinalizationHandler.js:9]: as for initial payment, we could rename it to withAddPaymentDetailsFinalizationHandler

File: src/containers/AddPaymentDetailsFinalizationHandler/AddPaymentDetailsFinalizationHandler.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  return function WithAddPaymentDetailsFinalizationHandler({
```
Comment (iwonakulacz) [inline src/containers/AddPaymentDetailsFinalizationHandler/AddPaymentDetailsFinalizationHandler.js:10]: and then there AddPaymentDetailsFinalizationHandler

File: src/containers/PaymentInfo/PaymentInfo.component.js [hunk 12]
```diff
@@ -164,15 +165,15 @@ const PaymentInfoFn = ({
 
     return () => {
       hideInnerPopup();
+      // replace with: hide payment info popup?
```
Comment (iwonakulacz) [inline src/containers/PaymentInfo/PaymentInfo.component.js:168]: I think that yes, as the payment info popup is the only popup in the PaymentInfo component

File: src/redux/finalizeAddPaymentDetailsSlice.js [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+  name: 'finalizeAddPaymentDetails',
+  initialState,
+  reducers: {},
+  extraReducers: {
+    [fetchFinalizeAddPaymentDetails.pending]: state => {
+      state.loading = true;
+    },
+    [fetchFinalizeAddPaymentDetails.fulfilled]: (state, { payload }) => {
+      state.loading = false;
+      state.paymentDetails = payload;
```
Comment (iwonakulacz) [inline src/redux/finalizeAddPaymentDetailsSlice.js:36]: lets set error to `null` in this case

File: src/translations/en/translations.json [hunk 14]
```diff
@@ -369,5 +369,7 @@
   "thank-you-page.text-with-payment-method": "Your purchase was successfully processed using {{paymentMethodName}}!",
   "thank-you-page.text": "Your purchase was successfully processed!",
   "thank-you-page.sub-text-with-price": "You have been charged {{currencySymbol}}{{totalAmount}}.",
-  "thank-you-page.manage-text": "You can manage your account from <1>here</1>."
+  "thank-you-page.manage-text": "You can manage your account from <1>here</1>.",
+  "update-payment-error-method": "We weren’t able to update your payment method. Please try again.",
+  "update-payment-error-details": "We weren’t able to update your payment details. Please try again using different payment method."
```
Comment (iwonakulacz) [inline src/translations/en/translations.json:374]: maybe we could change the keys to more like:
`update-payment-method.error`
`update-payment-method.refused`

File: src/components/UpdatePaymentDetailsPopup/Steps/Error.js (hunk 5 above)
Comment (jkorczakcleeng) [reply src/components/UpdatePaymentDetailsPopup/Steps/Error.js:55]: I agree that current behavior is better. But thanks to that I found one bug - when error action is undefined which cause error ;)

File: src/components/UpdatePaymentDetailsPopup/Steps/Error.js [hunk 15]
```diff
@@ -38,12 +38,12 @@ const Error = () => {
...
+            <Trans i18nKey="update-payment-error.method">
               We weren’t able to update your payment method. <br /> Please try
               again.
             </Trans>
           ) : (
-            <Trans i18nKey="update-payment-error-details">
+            <Trans i18nKey="update-payment-error.details">
               We weren’t able to update your payment details. <br /> Please try
               again using different payment method.
             </Trans>
```
Comment (iwonakulacz) [inline src/components/UpdatePaymentDetailsPopup/Steps/Error.js:49]: the keys here are still missleading for me. I would use `update-payment-details` (as a feature), then dot and type of error, so: 
`update-payment-details.refused`
`update-payment-details.error`

File: src/containers/PaymentInfo/PaymentInfo.container.js [hunk 16]
```diff
@@ -37,7 +37,7 @@ export const mapDispatchToProps = dispatch => {
     showInnerPopup: payload => {
       dispatch(showInnerPopup(payload));
     },
-    hideInnerPopup: () => {
+    hidePaymentInfoPopup: () => {
       dispatch(hideInnerPopup());
```
Comment (iwonakulacz) [inline src/containers/PaymentInfo/PaymentInfo.container.js:41]: as we replaced it with hidePaymentInfoPopup we could use here the new popup manager and new function to hide update payment details popup

**Review by pawelacio — APPROVED** [review]

**Pull request merged**